### PR TITLE
NEXT-9360 - add missing order delivery state transitions

### PR DESCRIPTION
**Pull request by jochenmanz**
### 1. What does this change do, exactly?
Add's state transitions from all order delivery states back to open.

### 3. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have squashed any insignificant commits
- [ ] I have created a [changelog file](https://github.com/shopware/platform/blob/master/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [ ] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: src/Core/Migration/Migration1600349343AddDeliveryStateTransitions.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    public function update(Connection $connection): void
+    {
+        $stateMachineId = $this->fetchOrderDeliveryStateId($connection);
+        $openStateId = $this->fetchOpenOrderDeliveryStateId($connection);
+        $missingStates = $this->fetchMissingOrderDeliveryStates($connection, $stateMachineId);
+
+        foreach ($missingStates as $missingState) {
+            $connection->insert(
```
Comment (OliverSkroblin) [inline src/Core/Migration/Migration1600349343AddDeliveryStateTransitions.php:24]: you should pre-validate if this not already exists

**Comment by shyim** [comment]
You mixed up single and double quotes

File: src/Core/Migration/Migration1600349343AddDeliveryStateTransitions.php (hunk 1 above)
Comment (jochenmanz) [reply src/Core/Migration/Migration1600349343AddDeliveryStateTransitions.php:24]: Its calculated inside fetchMissingOrderDeliveryStates, all present ropen transition states are exluded.

**Comment by jochenmanz** [comment]
> You mixed up single and double quotes

fixed

**Review by shyim — APPROVED** [review]

**Pull request merged**